Application/src/net/Decode.java

```diff
@@ -0,0 +1,142 @@
+import visuals.Draw;
+import visuals.ExecutionLog;
+
+import static visuals.DiagramView.tabPane;
+
+/**
+ * @author Pontus Laestadius
+ * @version 0.1
+ */
+class Decode {
+    String raw; // Raw string to be decoded.
+
+    /**
+     * @param string string to decode.
+     */
+    Decode(String string) {
+        raw = string;
+    }
+
+    /**
+     * Decodes the raw string and allocates it to it's associated diagram.
+     */
+    void execute() {
+        System.out.println(raw);
+
+        // If no string has been allocated, abort.
+        if (raw == null) return;
+
+        // Split the raw string in to fields.
+        int id_index = raw.indexOf(",");
+
+        String id_string = raw.substring(1, id_index);  // Retrieves the Diagram_ID.
+        int id = Integer.decode(id_string); // String to Integer.
+
+        if (raw.contains("simulation_finished")) { // If the simulation is finished.
+            write("Simulation finished"); // Write Simulation finished in the execution log.
+
+        } else if (raw.contains("print_information")) { // If it's a statement to only print.
+            write(raw); // Writes the raw data TODO replace with statement provided.
+            Net.push("{" + id + ", send_message}");
+
+        } else { // If it's a message or a list of classes.
+            int msg_start = raw.indexOf("["); // Finds the end of the message content.
+            int msg_end = raw.indexOf("]"); // Finds the start of the message content.
+            String msg = raw.substring(msg_start, msg_end); // Message content. as a substring.
+            String values = raw.replace(msg, ""); // Removes the message content from raw.
+            byte[] bytes = raw.getBytes(); // Gets the raw message as bytes.
+
+            // Identifies if it's a message or a new diagram by the next to last byte being a integer or not.
+            if (bytes[bytes.length-2] >= (byte)'0' && bytes[bytes.length-2] <= (byte)'9') { // Message
+
+                // Adds message to the DiagramView.
+                message(
+                        DiagramView.getDiagramViewInView().getDraw(),// TODO change if you want multiple diagrams.
+                        msg, // The message content.
+                        values.split(",")); // Split the fields remaining.
+            } else { // New Diagram
+                diagramClasses(msg);
+            }
+        }
+    }
+
+    /**
+     * Removes all the characters provided from the string provided.
+     * @param string the string to purify.
+     * @param characters The main card, "Exodia the Forbidden One",
+     *                   is one of the first five Effect Monsters
+     *                   ever to be released and the first card to
+     *                   have an alternative victory condition.
+     * @return a String without any of the characters provided.
+     */
+    private String removeCharactersFromString(String string, char... characters) {
+        StringBuilder sb = new StringBuilder();
```

Comment:
maybe it is just me but it is hard to read the code when i have to keep track of the very concise instance names, please make them more self explanatory e.g. UserName userName   instead of UserName un